assets/js/blocks/cart-checkout/mini-cart/edit.tsx

```diff
@@ -1,31 +1,107 @@
 /**
  * External dependencies
  */
-import { useBlockProps } from '@wordpress/block-editor';
+import {
+	InspectorControls,
+	useBlockProps,
+	getColorClassName,
+} from '@wordpress/block-editor';
 import type { ReactElement } from 'react';
 import { formatPrice } from '@woocommerce/price-format';
 import { CartCheckoutCompatibilityNotice } from '@woocommerce/editor-components/compatibility-notices';
+import { PanelBody, ToggleControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
 
 /**
  * Internal dependencies
  */
 import QuantityBadge from './quantity-badge';
 
-const MiniCartBlock = (): ReactElement => {
+interface Attributes {
+	isInitiallyOpen?: boolean;
+	transparentButton: boolean;
+	backgroundColor?: string;
+	textColor?: string;
+	style?: Record< string, Record< string, string > >;
+}
+
+interface Props {
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+}
+
+const MiniCartBlock = ( {
+	attributes,
+	setAttributes,
+}: Props ): ReactElement => {
+	const { transparentButton, backgroundColor, textColor, style } = attributes;
 	const blockProps = useBlockProps( {
-		className: 'wc-block-mini-cart',
+		className: classnames( 'wc-block-mini-cart', {
+			'is-transparent': transparentButton,
+		} ),
 	} );
 
+	/**
+	 * @todo Replace `getColorClassName` and manual style manipulation with
+	 * `useColorProps` once the hook is no longer experimental.
+	 */
+	const backgroundClass = getColorClassName(
+		'background-color',
+		backgroundColor
+	);
+	const textColorClass = getColorClassName( 'color', textColor );
+
 	const productCount = 0;
 	const productTotal = 0;
 
 	return (
 		<div { ...blockProps }>
-			<button className="wc-block-mini-cart__button">
+			<InspectorControls>
+				<PanelBody
+					title={ __(
+						'Button style',
+						'woo-gutenberg-products-block'
+					) }
+				>
+					<ToggleControl
+						label={ __(
+							'Use transparent button',
+							'woo-gutenberg-products-block'
+						) }
+						checked={ transparentButton }
+						onChange={ () =>
+							setAttributes( {
+								transparentButton: ! transparentButton,
+							} )
+						}
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<button
+				className={ classnames( 'wc-block-mini-cart__button', {
+					[ backgroundClass ]: backgroundClass,
+					[ textColorClass ]: textColorClass,
+				} ) }
```

Comment:
Nit, I think you can rewrite this like:
```suggestion
				className={ classnames(
					'wc-block-mini-cart__button',
					backgroundClass,
					textColorClass
				) }
```
Keeps the code a bit cleaner and gets rid of a couple of TypeScript warnings.